pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,411 @@
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineCluster{Instancetype,Preference} and directly used PVC
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --pvc=my-pvc`
+}
+
+func splitPrefixedName(prefixedName string) (prefix string, name string, err error) {
+	split := strings.Split(prefixedName, "/")
+
+	switch l := len(split); l {
+	case 1:
+		name = split[0]
+	case 2:
+		prefix = split[0]
+		name = split[1]
+	default:
+		return "", "", fmt.Errorf("invalid count %d of slashes in prefix/name", l)
+	}
+
+	if name == "" {
+		return "", "", errors.New("name cannot be empty")
+	}
+
+	return
+}
+
+func newVM(c *createVM) *v1.VirtualMachine {
+	runStrategy := v1.VirtualMachineRunStrategy(c.runStrategy)
+	return &v1.VirtualMachine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       v1.VirtualMachineGroupVersionKind.Kind,
+			APIVersion: v1.VirtualMachineGroupVersionKind.GroupVersion().String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: v1.VirtualMachineSpec{
+			RunStrategy: &runStrategy,
+			Template: &v1.VirtualMachineInstanceTemplateSpec{
+				Spec: v1.VirtualMachineInstanceSpec{
+					TerminationGracePeriodSeconds: &c.terminationGracePeriod,
+				},
+			},
+		},
+	}
+}
+
+func withRunStrategy(c *createVM, vm *v1.VirtualMachine) error {
+	for _, runStrategy := range runStrategies {
+		if runStrategy == c.runStrategy {
+			vmRunStrategy := v1.VirtualMachineRunStrategy(c.runStrategy)
+			vm.Spec.RunStrategy = &vmRunStrategy
+			return nil
+		}
+	}
+
+	return fmt.Errorf("unknown RunStrategy %s, supported RunStrategies are: %s", c.runStrategy, strings.Join(runStrategies, ", "))
+}
+
+func withInstancetype(c *createVM, vm *v1.VirtualMachine) error {
+	kind, name, err := splitPrefixedName(c.instancetype)
+	if err != nil {
+		return err
+	}
+
+	if kind != "" && kind != instancetype.SingularResourceName && kind != instancetype.ClusterSingularResourceName {
+		return fmt.Errorf("invalid instancetype kind %s", kind)
+	}
+
+	// If kind is empty we rely on the vm-mutator to fill in the default value VirtualMachineClusterInstancetype
+	vm.Spec.Instancetype = &v1.InstancetypeMatcher{
+		Name: name,
+		Kind: kind,
+	}
+
+	return nil
+}
+
+func withPreference(c *createVM, vm *v1.VirtualMachine) error {
+	kind, name, err := splitPrefixedName(c.preference)
+	if err != nil {
+		return err
+	}
+
+	if kind != "" && kind != instancetype.SingularPreferenceResourceName && kind != instancetype.ClusterSingularPreferenceResourceName {
+		return fmt.Errorf("invalid preference kind %s", kind)
+	}
+
+	// If kind is empty we rely on the vm-mutator to fill in the default value VirtualMachineClusterPreference
+	vm.Spec.Preference = &v1.PreferenceMatcher{
+		Name: name,
+		Kind: kind,
+	}
+
+	return nil
+}
+
+func withDataSource(c *createVM, vm *v1.VirtualMachine) error {
+	for _, dataSource := range c.dataSources {
```

Comment:
How is it possible to set multiple data sources? Is it possible to provide the same flag multiple times? Or separated by a certain character? Perhaps it should be also documented?

Reply:
Yes, that flag can be provided multiple times. I will add an example.